### PR TITLE
fix: reset lock on reset of redux store

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** `fix` 

https://www.notion.so/appbase/RS-initialQuerySyncTime-breaks-for-routed-apps-a6d2caedf7ab44adb59b48b4222e9290

https://www.loom.com/share/e597c1a932bd4c589a8f8ec068797227

### CODE REVIEW

File: src/actions/query.js [hunk 1]
```diff
@@ -568,6 +568,14 @@ export function executeQuery(
 										if (queryExecutionMap[query.id]) {
 											newQuery.execute = true;
 										}
+										if (
```
Comment (bietkul) [inline src/actions/query.js:571]: @mohdashraf010897 It is intentional that `type` is changing to `search` when `suggestion` type query is used with `result` and have `execute` as `false`. We should not change the type to `suggestion`.
IMO, we can remove this logic.

File: src/actions/query.js [hunk 2]
```diff
@@ -597,7 +605,7 @@ export function executeQuery(
 							}));
 						}, lockTime);
 					}
-					lock = true;
+					dispatch(updateStoreConfig({ lock: true }));
```
Comment (bietkul) [inline src/actions/query.js:615]: @mohdashraf010897 Can you confirm if it is a patch update, i.e. wouldn’t override existing properties?

**Review by bietkul — CHANGES_REQUESTED** [review]

File: src/actions/query.js (hunk 2 above)
Comment (mohdashraf010897) [reply src/actions/query.js:608]: Then the logic won't make a suggestions query for initial suggestions

File: src/actions/query.js (hunk 1 above)
Comment (mohdashraf010897) [reply src/actions/query.js:571]: @bietkul 
then suggestions query for initial suggestions won't be made

File: src/actions/query.js (hunk 2 above)
Comment (mohdashraf010897) [reply src/actions/query.js:615]: @bietkul 
yes, I can confirm it. Check screenshot below

image

**Pull request merged**